app/controllers/stories_controller.rb

```diff
@@ -85,6 +85,13 @@ def create
     end
   end
 
+  def history
+    @story = policy_scope(Story).try(:find, params[:story_id])
```

Comment:
I think you can remove the try and change the method to find_by, so if the record isn't found it will block in authorize method.

Also, isn't better to create a separated controller to this? This controllers seems to be huge right now.